pyrealm/demography/canopy.py

```diff
@@ -0,0 +1,133 @@
+"""Functionality for canopy modelling."""
+
+import numpy as np
+from numpy.typing import NDArray
+from scipy.optimize import root_scalar
+
+from pyrealm.demography.canopy_functions import (
+    calculate_relative_canopy_radius_at_z,
+    calculate_stem_projected_crown_area_at_z,
+    calculate_stem_projected_leaf_area_at_z,
+    solve_community_projected_canopy_area,
+)
+from pyrealm.demography.community import Community
+
+
+class Canopy:
+    """A class containing attributes of a canopy, including the structure.
+
+    at which canopy layers close. This also includes
+        ground level, which will capture all remaning crown and leaf area below the last
+        closed layer.
+    """
+
+    def __init__(
+        self,
+        community: Community,
+        canopy_gap_fraction: float,
+        layer_tolerance: float = 0.001,
```

Comment:
Perhaps worth outlining in the docstring this default value.

Reply:
We've actually got an issue about this (#270)  - @j-emberton pointed out that the documentation of defaults and optional values isn't consistent throughout the codebase.

Reply:
Not that this stops me fixing this specific instance 😄